grails-app/services/org/pih/warehouse/product/ProductService.groovy

```diff
@@ -1228,15 +1228,40 @@ class ProductService {
         return productInstance
     }
 
+    Attribute findExternalProductIdAttribute(String externalAttributeCode = null){
+        if (!externalAttributeCode) {
+            externalAttributeCode = grailsApplication.config.externalProductId.attribute.code
+        }
+        if (!externalAttributeCode) {
+            externalAttributeCode = "EXTERNAL_PRODUCT_ID"
+        }
+        Attribute attribute = Attribute.findByCode(externalAttributeCode)
+        return attribute
+    }
+
+    Product findProductByExternalId(String externalProductId){
```

Comment:
Make sure we check to see if this is null or empty. My guess is that we should throw an IllegalArgumentException?